src/kv/deserialise.h

```diff
@@ -118,7 +118,9 @@ namespace kv
 
       if (history)
       {
-        history->append(data);
+        history->append(
```

Comment:
These are the kind of changes required because we decided not to include this new header in the Merkle tree (see also `historical_queries.h`, `store.h`). In hindsight, while it wouldn't provide much security guarantee, I think it's pretty inelegant and I'm now keen to include it in the hash of the Merkle Tree. @achamayou, @eddyashton thoughts?

Reply:
+1, I also think the header should be included in the data. Makes everything much easier to read and interpret. I suppose the ledger would grow a little because of the copies of identical headers, but that's acceptable I think. I have no idea what I'm talking about, but it seems to me that it does at least provide a record of ill behaviour on the ledger if someone tries to play games with old version numbers (but that's probably not really necessary). I think it also means that ledger verification is less stateful because we don't need to remember headers/versions/flags of previous entries.

Reply:
I agree this is inelegant. The problem here is that we have some internal structure that we're accessing in lots of places. For argument's sake, I see 2 other approaches other than adding the header to the Merkle tree:
- Hide this indirection in a new type. We could return this type from `serialised` (rather than the raw `std::vector`), or it could just have a bunch of static functions that operate over a `std::vector`. So this would be something like `history->append(LedgerEntry::hashed_body(data))`, but with a better name for "the bit that goes in the Merkle tree" than `hashed_body`.
- Extract these internal bits _inside_ `history->append`. So `append` takes a `SerialisedLedgerEntry` (aka a `std::vector`, but to be clear this is an entire `Entry`), and skips the prefix. If we have raw `append`s elsewhere, we could do this with 2 functions (`append_raw` vs `append_from_entry`? Naming is hard).

Reply:
> I think it also means that ledger verification is less stateful because we don't need to remember headers/versions/flags of previous entries.

The primary motivation for the introduction of flags is to enable custom claims, in which case the ledger verification will necessarily require more interpretation (likely a small mini-tree for each entry), so this isn't going to stay simple for very long.

Since these flags will (among other things) communicate to an auditor how to build the tree, putting them inside is quite circular and does not obviously prevent bad behaviour.

Reply:
@eddyashton: Agreed that we could do this in a cleaner way if we decide to not include the headers in the Merkle tree. My concern is that you still have to remember to do this every time you receive an entry from the KV/ledger and want to append it to the tree (e.g. for historical queries), as opposed to having everything that comes out of the KV or read from the ledger as the input to the Merkle tree. Because of this, I'm still leaning towards including the header in the Merkle tree.

Reply:
As discussed with the broader team, let's include the prefix in what we hash. There isn't a material upside at the moment other than the code is a little cleaner, but we don't think there is a downside either.

Reply:
Done.